sirepo/package_data/static/js/sirepo.js

```diff
@@ -96,6 +99,58 @@ angular.element(document).ready(function() {
             : addTag(src, 'script', 'body', 'src', {'type': 'text/javascript', 'async': true});
     }
 
+    function loadFromPath(path, type, callback, errCallback, finallyCallback) {
+        let d = $.Deferred();
+        $.get(`${path}${SIREPO.SOURCE_CACHE_KEY}`, function (res) {
+            callback(res);
+        }, type)
+            .fail(function (res) {
+                if (errCallback) {
+                    errCallback(res);
+                }
+            })
+            .always(function (res) {
+                if (finallyCallback) {
+                    finallyCallback(res);
+                }
+                d.resolve();
+            });
+        return d.promise();
+    }
+
+    function loadBlock(b) {
+        let val = null;
+        let status = 500;
+        return loadFromPath(b.path, b.fileType, function (res) {
+            val = res;
+            status = 200;
+        }, function(res) {
+            // turns out "svg" (etc.) is not a valid AJAX type, so we'll get an error.
+            // However we can get a successful read anyway; if so assign the response text
+            status = res.status;
+            val = res.responseText;
+        }, function (res) {
+            if (status === 200) {
+                SIREPO.BLOCKS[b.name] = val;
+                if (b.blockType === 'panel') {
+                    buildPanel(b.name, val);
+                }
+                return;
+            }
+            throw new Error(`${status}: Failed to load block ${b.name}`);
```

Comment:
reverse this test. `if (!ok) { throw}`. You don't need `return`, and the complex code is less indented.

Reply:
60e7ccc